src/main/kotlin/io/github/corbym/dokker/dokker.kt

```
@@ -9,9 +10,55 @@ fun dokker(init: DokkerContainerBuilder.() -> Unit): DokkerContainer {
     return DokkerContainerBuilder().apply(init).build()
 }
 
+// Search order is
+// 1. process name set in code (if set, DokkerAutoProcessSearchResult.processName is not used)
+// 2. environment variable (if found in path)
+// 3. hardcoded "docker" (if found in path)
+// 3. hardcoded "podman" (if found in path)
+object DokkerAutoProcessSearchResult {
+    private fun checkEnvVar(): String? {
+        return try {
+            System.getenv("DOKKER_PROCESS")
+        } catch (e: Exception) {
```

what kind of exception can happen here?

---

> If a security manager exists, its checkPermission method is called with a RuntimePermission("getenv."+name) permission. This may result in a SecurityException being thrown. If no exception is thrown the value of the variable name is returned.
Obviously, if this were internal software, wouldn't bother, but since it's open. Maybe it is overkill. +💯

---

> > If a security manager exists, its checkPermission method is called with a RuntimePermission("getenv."+name) permission. This may result in a SecurityException being thrown. If no exception is thrown the value of the variable name is returned.
> > Obviously, if this were internal software, wouldn't bother, but since it's open. Maybe it is overkill. +💯

we should probably just leave the securityException to get thrown out of this call, its probably something the user would want to know about if it occurred so they could fix it. If we had some kind of NPE getting thrown if the _property_ didn't exist then fair enough, we would want to catch it. But a NPE is only thrown from this method if the name of the property passed in is null, and it never will be..

---

i mean, yeah, but then all commands will fail to resolve)) and you would get a message anyway that no providers found etc
that seems to be kind of error that we dont need to catch and let it bubble up to the top 🤷